### PR TITLE
15. Subscription sign-out notification observers and Design Feedback

### PR DESCRIPTION
**Pull request by afterxleep**
Task/Issue URL: https://app.asana.com/0/414235014887631/1206458412095677/f

**Description**:
- Design review feedback updates (copy, layout fixes, etc)
- Removes debug flags on alpha builds (cc @graeme)
- Add notification observers for signOut on notifications

**Steps to test this PR**:
1.  Validate The subscription restore page looks fine, compared to this reference:
Screenshot 2024-03-01 at 3 48 16 PM

2.  Whenever possible, leave settings page open, put app in background and come back after subscription is expired (Should refresh settings)

### CODE REVIEW

File: DuckDuckGo.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -9344,7 +9340,7 @@
 				ONLY_ACTIVE_ARCH = YES;
 				OTHER_LDFLAGS = "-ld_classic";
 				SDKROOT = iphoneos;
-				SWIFT_ACTIVE_COMPILATION_CONDITIONS = "DEBUG NETWORK_PROTECTION ALPHA SUBSCRIPTION";
```
Comment (afterxleep) [inline DuckDuckGo.xcodeproj/project.pbxproj:9383]: @graeme  👁️

File: DuckDuckGo/Subscription/ViewModel/SubscriptionSettingsViewModel.swift [hunk 2]
```diff
@@ -76,6 +79,14 @@ final class SubscriptionSettingsViewModel: ObservableObject {
         }
     }
     
+    private func setupNotificationObservers() {
+        signOutObserver = NotificationCenter.default.addObserver(forName: .accountDidSignOut, object: nil, queue: .main) { [weak self] _ in
```
Comment (miasma13) [inline DuckDuckGo/Subscription/ViewModel/SubscriptionSettingsViewModel.swift:83]: You need to also remove closure based observers e.g. on deinit

**Review by miasma13 — APPROVED** [review]

File: DuckDuckGo/SettingsViewModel.swift [hunk 3]
```diff
@@ -390,6 +393,15 @@ extension SettingsViewModel {
                 self?.state.subscription.canPurchase = !products.isEmpty
             }.store(in: &cancellables)
     }
+        
+    private func setupNotificationObservers() {
+        signOutObserver = NotificationCenter.default.addObserver(forName: .accountDidSignOut, object: nil, queue: .main) { [weak self] _ in
```
Comment (miasma13) [inline DuckDuckGo/SettingsViewModel.swift:400]: You need to also remove closure based observers e.g. on deinit

File: DuckDuckGo/SettingsViewModel.swift [hunk 4]
```diff
@@ -390,6 +393,15 @@ extension SettingsViewModel {
...
+            if #available(iOS 15.0, *) {
+                self?.signOutUser()
```
Comment (miasma13) [inline DuckDuckGo/SettingsViewModel.swift]: Doesn't it causes a loop?
`signOutUser()` -> `AccountManager().signOut()` -> posts `.accountDidSignOut`

File: DuckDuckGo/Subscription/ViewModel/SubscriptionSettingsViewModel.swift (hunk 2 above)
Comment (afterxleep) [reply DuckDuckGo/Subscription/ViewModel/SubscriptionSettingsViewModel.swift:83]: oh, yeah.  Missed it.  Good call.

File: DuckDuckGo/SettingsViewModel.swift [hunk 5]
```diff
@@ -397,7 +402,8 @@ extension SettingsViewModel {
     private func setupNotificationObservers() {
         signOutObserver = NotificationCenter.default.addObserver(forName: .accountDidSignOut, object: nil, queue: .main) { [weak self] _ in
             if #available(iOS 15.0, *) {
-                self?.signOutUser()
+                guard let strongSelf = self else { return }
+                Task { await strongSelf.setupSubscriptionEnvironment() }
```
Comment (miasma13) [inline DuckDuckGo/SettingsViewModel.swift:406]: Will it have any effect, the `setupSubscriptionEnvironment()` will return at first guard due to the lack of access token.

**Pull request merged**